### DISCUSSION TITLE
Learning Vim with lazyvim, having trouble with restoring sessions

### DISCUSSION THREAD

**Discussion by biglari** [LazyVim/LazyVim#1441] [Q&A]
Hi, I am new to Vim and decided to dive in with LazyVim as a starting config. I set up my environment, changed some configs and am starting to get a hang of things. Recently I started to use Neovim for more serious development work, which led me to rely on saving and restoring sessions. 

Every time I started a session, the layout would be different from when I closed Neovim. Specifically, it seems Neo-Tree is either not visible, or in a split which was previously occupied by another buffer. On top of that, bufferline would show tabs for buffers which I had never opened, like [No Name], or the current directory (if I clicked on that one, it would display Neo-tree). 

I realize I am very new, and may have either installed or configured something incorrectly. I reinstalled lazyvim starter on a different user to see if any of my configuration changes were at fault, but it seems the fresh install has the same Problem. Is there something I am doing wrong? I was not able to find any other discussion or issue with this same problem, so perhaps I am doing something wrong?

Right before closing neovim:
![before_close](https://github.com/LazyVim/LazyVim/assets/5347067/31f4b3cc-77e6-4698-9d30-4672a76dbae0)

After restoring Session:
![after_restore](https://github.com/LazyVim/LazyVim/assets/5347067/25cf54e5-6dc0-48f9-bd52-b892b7251319)

Would appreciate any pointers and help.

**Comment by dpetka2001** [comment]
For me it doesn't save the state of `neo-tree` because it's buffer is not listed. But it restores the other files and even the splits like normal.

**Reply by biglari** [reply]
I would prefer it not loading neo-tree to it disrupting my splits and adding new buffers to bufferline. I wonder why its acting different for me?

**Comment by AddictArts** [comment]
This may be an issue with the plugin that Lazyvim uses https://github.com/folke/persistence.nvim. May be helpful to post a topic there. The session saving is somewhat a strange beast. It works for the most part, but I find it doesn't have the intuitive behavior. It doesn't always do what you think it would. Example closing a buffer only to restore the session and it be there. Hopefully at some point your discussion item is addressed in some way.
